eth/blockwatch/block_watcher.go

```diff
@@ -146,6 +146,27 @@ func (w *Watcher) InspectRetainedBlocks() ([]*MiniHeader, error) {
 	return w.stack.Inspect()
 }
 
+func (w *Watcher) syncToLatestBlock() error {
+	w.Lock()
+	defer w.Unlock()
+	lastBlock, err := w.client.HeaderByNumber(nil)
+	if err != nil {
+		return err
+	}
+
+	latestHeader, err := w.stack.Peek()
+	if err != nil {
+		return err
+	}
+
+	for i := lastBlock.Number; i.Cmp(latestHeader.Number) == 0; i = i.Add(i, big.NewInt(1)) {
```

Comment:
There are a few issues with this loop:

- The conditional needs to be updated here. This loop would never run when latestHeader.Number != i because `i.Cmp(latestHeader.Number) == 0` evaluates to false on the first iteration
- The loop counter is not being initialized properly. lastBlock.Number will actually be the latest block number and latestHeader.Number will actually be the last block number stored (the variable names should be updated to clarify the value they refer to). So, if lastBlock.Number > latestHeader.Number, this loop would never run even if the conditional issue mentioned above was fixed

Let's fix this and make sure to at the very least manually test it as well.

Reply:
manually tested on rinkeby with a block polling interval of 1 minute and confirmed we process 3-4 blocks every interval